### DISCUSSION TITLE
sqls lsp server has been archived; is there an actively maintained alternative?

### DISCUSSION THREAD

**Discussion by matt-y** [emacs-lsp/lsp-mode#3971] [Announcements]
[The sqls lsp server has been archived](https://github.com/lighttiger2505/sqls). This is[ currently listed as the lsp server option for SQL in the languages index.](https://emacs-lsp.github.io/lsp-mode/page/lsp-sqls/). 

Is there a known alternative that is actively maintained that lsp-mode could switch over to?

**Comment by yyoncho** [comment]
There is one but only for mssql server. Not sure if there are other options.

**Reply by matt-y** [reply]
[This could be a potential option](https://github.com/joe-re/sql-language-server) but I have not examined this repo in great detail - same license as `sqls`.

**Reply by phuhl** [reply]
Was hoping I could use sql-language-server (what you linked) with LSP as it seems actively maintained and supports 3 important DB engines. Could you give me any pointers on how this might work with lsp-mode today, even though not officially supported? Thx

**Reply by dschrempf** [reply]
I think development of `sqls` is again active: https://github.com/sqls-server/sqls

**Comment by pataquets** [comment]
@matt-y: As @dschrempf already pointed out, it is well alive at https://github.com/sqls-server/sqls
Fancy to close the discussion? I could not vote @dschrempf's comment as solution, thou.

**Comment by jcs090218** [comment]
`sqls` server is again active: https://github.com/sqls-server/sqls

There are two more language servers as alternatives:

- https://github.com/emacs-lsp/lsp-mode/blob/master/clients/lsp-sql.el
- https://github.com/emacs-lsp/lsp-mode/blob/master/clients/lsp-postgres.el
